File: include/restricted-extras-poppler/libpoppler_version.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// ### [Upstream resources](https://poppler.freedesktop.org/api/qt6/classPoppler_1_1Version.html)
pub const poppler__version = struct {
    /// ### [Upstream resources](https://poppler.freedesktop.org/api/qt6/classPoppler_1_1Version.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn String(allocator: std.mem.Allocator) []const u8 {
        var _str = qtc.Poppler__Version_String();
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("poppler__version.String: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### [Upstream resources](https://poppler.freedesktop.org/api/qt6/classPoppler_1_1Version.html)
    ///
    pub fn Major() u32 {
        return qtc.Poppler__Version_Major();
    }

    /// ### [Upstream resources](https://poppler.freedesktop.org/api/qt6/classPoppler_1_1Version.html)
    ///
    pub fn Minor() u32 {
        return qtc.Poppler__Version_Minor();
    }

    /// ### [Upstream resources](https://poppler.freedesktop.org/api/qt6/classPoppler_1_1Version.html)
    ///
    pub fn Micro() u32 {
        return qtc.Poppler__Version_Micro();
    }
};

File: src/restricted-extras-poppler/libpoppler_version.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Poppler__Version
#include <QString>
#include <QByteArray>
#include <cstring>
#include <poppler-version.h>
#include "libpoppler_version.h"
#include "libpoppler_version.hxx"

libqt_string Poppler__Version_String() {
    QString _ret = Poppler::Version::string();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

unsigned int Poppler__Version_Major() {
    return Poppler::Version::major();
}

unsigned int Poppler__Version_Minor() {
    return Poppler::Version::minor();
}

unsigned int Poppler__Version_Micro() {
    return Poppler::Version::micro();
}
